actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -16,13 +16,65 @@ def initialize(format)
       super "No renderer defined for format: #{format}"
     end
   end
+  # RFC: Consider documenting or testing MissingRenderer
+  #
+  # See <tt>Responder#api_behavior</tt>
+  # https://github.com/plataformatec/responders/blob/f0e093d01e799d8b67952c013daf8808e9ea02db/lib/action_controller/responder.rb#L209
+  #
+  #   # This is the common behavior for formats associated with APIs, such as :xml and :json.
+  #   def api_behavior
+  #     raise MissingRenderer.new(format) unless has_renderer?
+  #   end
+  #
+  #   # Check whether the necessary Renderer is available
+  #   def has_renderer?
+  #     Renderers::RENDERERS.include?(format)
+  #   end
+
+  # See <tt>Renderers.add_serializer</tt>
+  def self.add_serializer(key, &block)
+    Renderers.add_serializer(key, &block)
+  end
+
+  # See <tt>Renderers.remove_serializer</tt>
+  def self.remove_serializer(key)
+    Renderers.remove_serializer(key)
+  end
+
+  # See <tt>Renderers::MissingRenderer</tt>
+  class MissingSerializer < LoadError
+    def initialize(format)
+      super "No serializer defined for format: #{format}"
+    end
+  end
 
   module Renderers
     extend ActiveSupport::Concern
 
+    # A Set containing renderer names that correspond to available renderer procs.
+    # Default values are <tt>:json</tt>, <tt>:js</tt>, <tt>:xml</tt>.
+    RENDERERS = Set.new
```

Comment:
Was thinking an interesting style would be to define renderers per mime_type or format e.g. it would build something like

Given:

```ruby
content_type = 'application/json'
mime_type = Mime::Type.lookup(content_type)
format = :json
default_renderer = :json
```

then

```ruby
RENDERERS = {
  json: {  # 'format', e.g. `render json: model`
    content_type: Mime[:json], # each format has an associated mime type
    serializers: {
      default: "_render_with_renderer_json(json, options)" # this is basically the default json renderer.  Maybe it should be a callable, but this is the idea
    }
  }
}
```

Then, you could `ActionController::Renderers.add :json, :custom_serializer { |json, options| whatever }` and `render json: model, serializer: :custom_serializer` or maybe even `use_serializer json: :custom_serializer`.

To render JSON API it might make sense to add a `json_api` renderer and serializer e.g.

```ruby
RENDERERS = {
  json: {  } # etc, see above
  json_api: {  # 'format', e.g. `render json_api: model`
    content_type: Mime[:json-api], # each format has an associated mime type
    serializers: {
      default: "ActiveModel::SerializableResource.new(json, options)" # this is basically the ActiveModelSerializers serializer
    }
  }
}
```

But I'm not sure I'd want to change my renders from the `render json: model`, so maybe consider that the `json` format, and it uses the `json` renderer (calls `as_json` and sets `content_type`) and `serializer: :json_api` (representation of 'model' that be as_json, if not a string)